js/hooks/useNfc.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+
+export type NfcResult =
+  | { status: "reading" }
+  | { status: "success"; data: string }
+  | { status: "cancelled" }
+  | { status: "error"; error: Error };
+
+export const useNfc = (cancel: AbortController): NfcResult => {
+  const [result, setResult] = useState<NfcResult>({ status: "reading" });
```

Comment:
It seems incorrect that on mount you'd start in `reading` state, even if `NDEFReader` is either unavailable or not working properly.

Reply:
I was working off of the Glides NFC code and included the `nfcSupported` utility function so that we can determine whether to route the user to the NFC-based flow or not, so this hook shouldn't be used from the non-NFC-flow components. That said, I could also modify things such that the `NfcResult` type instead has an `"nfcUnsupported"` status or something and handle that case in the hook itself. What do you think of that?

Reply:
Okay, I actually gave it a shot in faf48e6 - let me know what you think.

Reply:
Because of what we discussed re: hooks, I think this is an improvement!

Reply:
Yeah, the way Glides does it, the NFC tap functionality is associated with a component, and while you can't conditionally call a hook you _can_ conditionally render a component. But I guess we shouldn't lock ourselves into this approach in Orbit.